### PR TITLE
Refactor jobs

### PR DESCRIPTION
**Pull request by natalie-white-aws**
Refactor legacy Job framework to enforce parameters with interface contracts instead of validations. Left legacy job and job-executable files intact (except for constants locations and refactors from classes to enums) for teams' education; will remove later. Need to create pyspark-etl-job tests but pushing for PR to unblock the rest of the team.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+ *  override to 3.0. The following ETL features are enabled by default:
+ *  —enable-metrics, —enable-spark-ui, —enable-continuous-cloudwatch-log.
+ *  You can find more details about version, worker type and other features
+ *  in Glue's public documentation.
+ */
+
+/**
+ * Properties for creating a Python Spark ETL job
+ */
+export interface PySSparkEtlJobProperties extends JobProperties {
```
Comment (askarserikov) [inline packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts:24]: If not intentional, probably should be PySparkEtlJobPropoerties?

File: packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+   * S3 URL where additional python dependencies are located
+   */
+  readonly extraPythonFiles?: string[];
+
+}
+
+/**
+ * A Python Spark ETL Glue Job
+ */
+export class PySparkEtlJub extends Job {
```
Comment (askarserikov) [inline packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts:47]: Probably a typo as well, PySparkEtlJob

File: packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    // Conbine command line arguments into a single line item
+    const defaultArguments = {
+      ...execuatbleArgs,
+      ...continuousLoggingArgs,
+      ...profilingMetricsArgs,
+      ...sparkUIArgs?.args,
+      ...this.checkNoReservedArgs(props.defaultArguments),
+    };
+
+    if ((!props.workerType && props.numberOrWorkers !== undefined) || (props.workerType && props.numberOrWorkers === undefined)) {
```
Comment (askarserikov) [inline packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts:102]: How do we decide when to push these validation checks to the base job class/job executable or in each job class? I have several checks like this within Ray class, thinking whether we should move them all to one place

File: packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+      description: props.description,
+      role: this.role.roleArn,
+      command: {
+        name: JobType.ETL,
+        scriptLocation: this.codeS3ObjectUrl(props.script),
+        pythonVersion: PythonVersion.THREE_NINE,
+      },
+      glueVersion: props.glueVersion ? props.glueVersion : GlueVersion.V4_0,
+      workerType: props.workerType ? props.workerType : WorkerType.G_2X,
+      numberOfWorkers: props.numberOrWorkers,
```
Comment (askarserikov) [inline packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts:117]: No defaults for numberOfWorkers and maxRetries or do we enforce them elsewhere?

File: packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+      maxRetries: props.maxRetries,
+      executionProperty: props.maxConcurrentRuns ? { maxConcurrentRuns: props.maxConcurrentRuns } : undefined,
+      //notificationProperty: props.notifyDelayAfter ? { notifyDelayAfter: props.notifyDelayAfter.toMinutes() } : undefined,
```
Comment (askarserikov) [inline packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts:120]: what do we do with this one?

File: packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts (hunk 3 above)
Comment (natalie-white-aws) [reply packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts:102]: The validations are different for each job type - see the RFC for specifics.

File: packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts (hunk 2 above)
Comment (natalie-white-aws) [reply packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts:47]: Yep, typo, will fix.

File: packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts (hunk 1 above)
Comment (natalie-white-aws) [reply packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts:24]: Yep, typo, will fix.

File: packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts (hunk 5 above)
Comment (natalie-white-aws) [reply packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts:120]: I have an ask out to Parag on the Glue service team on this one.

File: packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts (hunk 4 above)
Comment (natalie-white-aws) [reply packages/@aws-cdk/aws-glue-alpha/lib/jobs/pyspark-etl-job.ts:117]: I have an ask to Parag on the Glue service team for this one - will update when i get an answer

**Comment by natalie-white-aws** [comment]
Merging job refactor to main to unblock team progress

**Pull request merged**